### PR TITLE
Add metadata to compiled nin executables

### PR DESCRIPTION
**Pull request by aleksanb**
awesomee

:D

### CODE REVIEW

File: nin/backend/compress.js [hunk 1]
```diff
@@ -92,8 +94,14 @@ function compress(projectPath, payload, callback) {
           'demo=bootstrap({layers:layers, camerapaths:camerapaths, onprogress: ONPROGRESS, oncomplete: ONCOMPLETE});' +
         '}' +
       '}' +
-    '}</script><canvas class=hide height='+ height + ' width=' + width + '></canvas><img src=# onload=z()><!--';
+    '}</script>' +
```
Comment (sigvef) [inline nin/backend/compress.js]: What does this change do?

File: nin/backend/compress.js [hunk 2]
```diff
@@ -92,8 +94,14 @@ function compress(projectPath, payload, callback) {
...
+    '<canvas class=hide height='+ height + ' width=' + width + '></canvas><img src=# onload=z()><!--';
   var htMlChunk = chunk('htMl', new Buffer(html));
+  var metadataChunk = chunk('iTXt', new Buffer(
```
Comment (sigvef) [inline nin/backend/compress.js]: `iTXt` chunks should probably follow the format in the spec, see http://www.libpng.org/pub/png/spec/1.2/PNG-Chunks.html (4.2.3.3. iTXt International textual data). You can use positiveNumberToBytes to get correct widths for the different fields.

Comment (aleksanb) [inline nin/backend/compress.js]: Oh cool! I'll use all the appropriate fields.

File: nin/backend/compress.js (hunk 1 above)
Comment (aleksanb) [inline nin/backend/compress.js]: I found it easier to read on the next line.
Should i move it back?

Comment (sigvef) [inline nin/backend/compress.js]: Ok, I wasn't sure if it was just a formatting thing or there was more meaning to it.

**Comment by aleksanb** [comment]
Ping

File: nin/backend/compile.js [hunk 3]
```diff
@@ -63,9 +64,21 @@ var compile = function(projectPath, options) {
         fs.writeFileSync(projectPath + '/bin/' + filename, data);
       });
     }
+
+    var settings = projectSettings.load(projectPath);
+    var metadata = [
```
Comment (sigvef) [inline nin/backend/compile.js]: Consider:
```
var metadata = {
  Title: settings.title,
  Author: settings.authors.join(', '),
...
```

:)

Comment (sigvef) [inline nin/backend/compile.js]: (unless they have to be in that order for some reason)

File: nin/backend/compress.js [hunk 4]
```diff
@@ -92,7 +93,13 @@ function compress(projectPath, payload, callback) {
           'demo=bootstrap({layers:layers, camerapaths:camerapaths, onprogress: ONPROGRESS, oncomplete: ONCOMPLETE});' +
         '}' +
       '}' +
-    '}</script><canvas class=hide height='+ height + ' width=' + width + '></canvas><img src=# onload=z()><!--';
+    '}</script>' +
+    '<canvas class=hide height='+ height + ' width=' + width + '></canvas><img src=# onload=z()><!--';
+
+
+  var metadataChunks = Buffer.concat(metadata.map(function(data) {
+    return chunk('tEXt', new Buffer(data[0] + '\0' + data[1]));
```
Comment (sigvef) [inline nin/backend/compress.js]: Perhaps use iTXt, since it supporst utf-8 instead of only latin-1.

File: nin/backend/compile.js (hunk 3 above)
Comment (aleksanb) [inline nin/backend/compile.js]: It's mainly in order for the metadata tEXt to look nicer with the title first. When generating the OG tags I think I might switch to a dictionary, as i need to reference the authors field directly then.

File: nin/backend/compress.js (hunk 4 above)
Comment (aleksanb) [inline nin/backend/compress.js]: Turns out implementing iTXt support is rather difficult, and finding a tool that's able to show those segments even more so.
I suggest we wait until a need for it appears.

**Pull request merged**